kit/transport/http/middleware_test.go

```diff
@@ -2,15 +2,113 @@ package http
 
 import (
 	"net/http"
+	"net/http/httptest"
 	"path"
 	"testing"
 
 	"github.com/influxdata/influxdb/v2/kit/platform"
+	"github.com/influxdata/influxdb/v2/kit/prom"
+	"github.com/influxdata/influxdb/v2/kit/prom/promtest"
+	"github.com/prometheus/client_golang/prometheus"
+	"go.uber.org/zap/zaptest"
 
 	"github.com/influxdata/influxdb/v2/pkg/testttp"
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 )
 
+func TestMetrics(t *testing.T) {
+	labels := []string{"handler", "method", "path", "status", "response_code", "user_agent"}
+
+	nextHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.URL.Path == "/" {
+			w.WriteHeader(http.StatusOK)
+			return
+		}
+
+		if r.URL.Path == "/serverError" {
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+
+		if r.URL.Path == "/redirect" {
+			w.WriteHeader(http.StatusPermanentRedirect)
+			return
+		}
+
+		w.WriteHeader(http.StatusNotFound)
+	})
+
+	tests := []struct {
+		name          string
+		reqPath       string
+		wantCount     int
+		labelResponse string
+		labelStatus   string
+	}{
+		{
+			name:          "counter increments on OK (2XX) ",
+			reqPath:       "/",
+			wantCount:     1,
+			labelResponse: "200",
+			labelStatus:   "2XX",
+		},
+		{
+			name:          "counter does not increment on not found (4XX)",
+			reqPath:       "/badpath",
+			wantCount:     0,
+			labelResponse: "404",
+			labelStatus:   "4XX",
+		},
+		{
+			name:          "counter increments on server error (5XX)",
+			reqPath:       "/serverError",
+			wantCount:     1,
+			labelResponse: "500",
+			labelStatus:   "5XX",
+		},
+		{
+			name:          "counter does not increment on redirect (3XX)",
+			reqPath:       "/redirect",
+			wantCount:     0,
+			labelResponse: "500",
+			labelStatus:   "5XX",
```

Comment:
Should `labelResponse` and `labelStatus` here be `300` and `3XX`? If the value doesn't affect the test result, could they be omitted from the declaration here to avoid confusion?

Reply:
Yep for test cases where the `wantCount` is `0` it doesn't even need those so I went ahead and deleted them. They should have been `300` and `3XX` but it was some bad copy/paste that doesn't matter anyway!